status_and_problems.tex

```diff
@@ -0,0 +1,134 @@
+
+A GitHub package may be included in more than one SW product.
+This make impossible to apply in a consistent way the release procedure to other SW products. It can be applied only to release the science pipelines.
+
+Follows an example, that explains why the procedure can't be applied to a different SW product.
+
+Product A is composed by 100 Git packages.
+
+Product B is composed by 80 Git packages.
+
+70 Git packages are shared by both products.
+
+When release 1.0 is done on product A, all 100 packages will be tagged, and corresponding Eups packages are created. This will include the 70 shared packages.
+
+One week lather, the release 1.0 should be done on product B. 
+Some of the 70 shared packages may have been updated for the need of the forthcoming 1.0 product B release.
+This imply that for some packages, release 1.0 of product A, is different from release 1.0 of product B.
+
+Requirement: each GitHub package shall be included in only one software product.
+
+Requirement: each GitHub package used for building, unit testing and packaging the SW Product shall not be included in the SW product itself, but versioned separatelly.
+
+
+\subsubsection{Code Fragmentation} \label{sec:problemCode}
+
+The high number of package.
+
+This problem imply:
+
+\begin{itemize}
+\item increase the build time: each Git package need to be build each time a change is introduced in one of the packages composing the SW product.
+\item increase the release time: all Git packages need to be released each time.
+\item increase the failure probability: the tooling my be affected by network glitch or similar technical issues. This may lead to the failure of the build/release process in a non deterministic way.
+\end{itemize}
+
+Moving 3rd party libraries to conda environment, will reduce this problem. 
+However this requires proper management of the conda environment.
+
+Requirement: conda environment shall be managed properly (activity already started).
+
+Requirement: the number of Git packages shall be kept low. DM-CCB shall approve each time a new package is introduced, since this has an impact on the maintainability of the system.
+
+
+\subsubsection{Binaries Persistence} \label{sec:problemPersistence}
+
+Each time we do a build, all git packages and rebuild, if not already build locally.
+
+In reality, once a package has been build, and the corresponding binaries have been made available, for a specific platform (linux, osx, etc), in the package repository (\url{https://eups.lsst.codes/}),
+it shall be possible for all the downstream users, to use it without build it again.
+
+A GitHub package should be rebuild, only in case its source code has changed, or one of its dependencies has changed (in case of semantic versioning, only for breaking changes that increase the major version).
+
+Requirement: It shall be possible to resolve the package binaries from \url{https://eups.lsst.codes/} if available, instead of building them from the source code.
```

Comment:
Ok here I agree -- or we put them all in conda ?   But where do we do the individual builds ?

Point out you still need some process like the current build which occasionally checks everything works together ..
Although we could assume things have gotten better and with proper dependency management this may be ok.  Still need a build system though ..

Reply:
Well. The scope of this TN is to approach the problem in a smooth way, without causing too many troubles to people that are used to a system for many many years.
Conda is a long term target.

Of course we need a build system part of the solution. We may keep what we have, and just change it a bit... or change everything and go to conda. But I don't think it is good to mention solutions here.